core/Access.php

```diff
@@ -241,11 +246,12 @@ protected function reloadAccessSuperUser()
      */
     public function setSuperUserAccess($bool = true)
     {
+        $this->hasSuperUserAccess = (bool) $bool;
+
         if ($bool) {
-            $this->reloadAccessSuperUser();
+            $this->makeSureLoginNameIsSet();
```

Comment:
You're right, when super user access is set, only the `'superuser'` element in idsitesByAccess or whatever is used.